sentry_streams/sentry_streams/examples/billing.py

```diff
@@ -1,50 +1,87 @@
-from sentry_streams.examples.billing_buffer import OutcomesBuffer
-from sentry_streams.pipeline.function_template import KVAggregationBackend
-from sentry_streams.pipeline.pipeline import (
-    Aggregate,
+from sentry_streams.pipeline import (
     Map,
-    Pipeline,
-    StreamSource,
+    Reducer,
+    streaming_source,
 )
+from sentry_streams.pipeline.chain import StreamSink
+from sentry_streams.pipeline.function_template import KVAccumulator
+from sentry_streams.pipeline.message import Message
 from sentry_streams.pipeline.window import TumblingWindow
 
 Outcome = dict[str, str]
 
 
-def build_outcome(value: str) -> Outcome:
-    d: Outcome = json.loads(value)
+class OutcomesBuffer(KVAccumulator[Message[Outcome]]):
+    """
+    An accumulator which adds outcomes data to a PendingBuffer.
+    Upon the closing of a window, the Buffer is flushed to a
+    sample backend (the OutcomesBackend). As of now this backend
+    is not a mocked DB, it is a simple hash map.
+    """
 
-    return d
+    def __init__(self, outcomes_dict: Optional[dict[str, int]] = None) -> None:
+        if outcomes_dict:
+            self.map: dict[str, int] = outcomes_dict
 
+        else:
+            self.map = {}
 
-# pipeline: special name
-pipeline = Pipeline()
+    def add(self, message: Message[Outcome]) -> Self:
+        value = message.payload
+        outcome_type = ""
 
-source = StreamSource(
-    name="myinput",
-    ctx=pipeline,
-    stream_name="events",
-)
+        if "state" in value:
+            outcome_type += value["state"]
+
+        if "data_cat" in value:
+            outcome_type += "-" + value["data_cat"]
+
+        if outcome_type in self.map:
+            self.map[outcome_type] += 1
+
+        else:
+            self.map[outcome_type] = 1
+
+        return self
+
+    def get_value(self) -> dict[str, int]:
+        return self.map
+
+    def merge(self, other: Self) -> Self:
+
+        first = self.map
+        second = other.map
+
+        for outcome_key in second:
+            if outcome_key in first:
+                first[outcome_key] += second[outcome_key]
+
+            else:
+                first[outcome_key] = second[outcome_key]
+
+        self.map = first
+
+        return self
+
+
+def build_outcome(value: Message[bytes]) -> Outcome:
+    d: Outcome = json.loads(value.payload.decode("utf-8"))
+
+    return d
 
-map = Map(
-    name="mymap",
-    ctx=pipeline,
-    inputs=[source],
-    function=build_outcome,
-)
 
-# A sample window.
-# Windows are assigned 3 elements.
-reduce_window = TumblingWindow(window_size=3)
-
-reduce: Aggregate[int, Outcome, dict[Any, Any]] = Aggregate(
-    name="myreduce",
-    ctx=pipeline,
-    inputs=[map],
-    window=reduce_window,
-    aggregate_func=OutcomesBuffer,
-    aggregate_backend=KVAggregationBackend(),  # NOTE: Provided by the platform
+pipeline = (
+    streaming_source(
+        name="myinput",
+        stream_name="events",
+    )
+    .apply("mymap", Map(function=build_outcome))
```

Comment:
Please use a `Parser` applier rather than a custom map. https://getsentry.github.io/streams/build_pipeline.html#sentry_streams.pipeline.chain.Parser

We have the outcome schema in the schema registry so it should just work https://github.com/getsentry/sentry-kafka-schemas/blob/main/schemas/outcomes.v1.schema.json

As people will refer to these example, we should follow the practices we want to recommend